### PR TITLE
Update samples to setup kafka simply

### PR DESCRIPTION
**Pull request by dbgsprw**
- Add docker-compose.yml
- Update README.adoc

Suggest an explanation about simple setting up kafka on the sample.

### CODE REVIEW

File: samples/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+version: '2'
+services:
+  zookeeper:
+    image: wurstmeister/zookeeper
```
Comment (daehokimm) [inline samples/docker-compose.yml:4]: IMHO, [confluentinc/cp-zookeeper](https://hub.docker.com/r/confluentinc/cp-zookeeper) & [confluentinc/cp-kafka](https://hub.docker.com/r/confluentinc/cp-kafka) image seems to be more maintainable. and if the images are changed, the options will have to change too.

**Review by daehokimm — COMMENTED** [review]
IMHO, [confluentinc/cp-zookeeper](https://hub.docker.com/r/confluentinc/cp-zookeeper) & [confluentinc/cp-kafka](https://hub.docker.com/r/confluentinc/cp-kafka) image seems to be more maintainable. and if the images are changed, the options will have to change too.

File: samples/docker-compose.yml (hunk 1 above)
Comment (dbgsprw) [reply samples/docker-compose.yml]: Thanks for your opinion.

**Comment by artembilan** [comment]
I don't think it is this project responsibility to bias to some specific Docker image.
It is really out of this project scope how your Kafka Broker is provided.

So, here is my 👎 for this suggestion.

Thanks

**Comment by dbgsprw** [comment]
> I don't think it is this project responsibility to bias to some specific Docker image.
> It is really out of this project scope how your Kafka Broker is provided.
> 
> So, here is my 👎 for this suggestion.
> 
> Thanks

I fully agree with your opinion on project responsibility, even though I may wish there was an easy way to execute the sample.

Then, what do you think about adding the default Kafka Broker used by this sample to README.md?

Thanks.

**Comment by artembilan** [comment]
Well, these samples are fully based on Spring Boot where a default `bootstrapServer` is definitely this one: `localhost:9092`.

So, it is fully up to end-user to ensure that Kafka broker is on that port.
I probably would start from regular Apache Kafka tutorial where they recommend to download Kafka and start it locally. 
And then I go to Spring solution to see what is there over the standard client. One more time: "client". There is nothing broker-specific in this project therefore we don't take responsibility what and how broker is provided.

With your suggestion I have to install Docker somehow, which doesn't work for me on Windows.
So, what is a simpler way to go then? Probably just leave a decision up to end-user without forcing them for some opinion which is not relevant to this project.

**Comment by dbgsprw** [comment]
> Well, these samples are fully based on Spring Boot where a default `bootstrapServer` is definitely this one: `localhost:9092`.
> 
> So, it is fully up to end-user to ensure that Kafka broker is on that port.
> I probably would start from regular Apache Kafka tutorial where they recommend to download Kafka and start it locally.
> And then I go to Spring solution to see what is there over the standard client. One more time: "client". There is nothing broker-specific in this project therefore we don't take responsibility what and how broker is provided.
> 
> With your suggestion I have to install Docker somehow, which doesn't work for me on Windows.
> So, what is a simpler way to go then? Probably just leave a decision up to end-user without forcing them for some opinion which is not relevant to this project.

@artembilan 
Thank you for your time and good opinion. I think it's important whether `localhost:9092` is predictable to end-user.
In your opinion it is predictable. I'll close this PR soon.

Thank you very much.

**Comment by garyrussell** [comment]
We should add test cases with the embedded kafka broker - we are always getting questions about testing with the embedded broker in Boot applications.

**Pull request closed without merging**